irohad/main/application.cpp

```diff
@@ -74,21 +75,23 @@ static constexpr iroha::consensus::yac::ConsistencyModel
 /**
  * Configuring iroha daemon
  */
-Irohad::Irohad(const std::string &block_store_dir,
-               const std::string &pg_conn,
-               const std::string &listen_ip,
-               size_t torii_port,
-               size_t internal_port,
-               size_t max_proposal_size,
-               std::chrono::milliseconds proposal_delay,
-               std::chrono::milliseconds vote_delay,
-               std::chrono::minutes mst_expiration_time,
-               const shared_model::crypto::Keypair &keypair,
-               std::chrono::milliseconds max_rounds_delay,
-               size_t stale_stream_max_rounds,
-               logger::LoggerManagerTreePtr logger_manager,
-               const boost::optional<GossipPropagationStrategyParams>
-                   &opt_mst_gossip_params)
+Irohad::Irohad(
+    const std::string &block_store_dir,
+    const std::string &pg_conn,
+    const std::string &listen_ip,
+    size_t torii_port,
+    size_t internal_port,
+    size_t max_proposal_size,
+    std::chrono::milliseconds proposal_delay,
+    std::chrono::milliseconds vote_delay,
+    std::chrono::minutes mst_expiration_time,
+    const shared_model::crypto::Keypair &keypair,
+    std::chrono::milliseconds max_rounds_delay,
+    size_t stale_stream_max_rounds,
+    std::vector<std::unique_ptr<shared_model::interface::Peer>> initial_peers,
```

Comment:
in this implementation this vector is an indirect switch for alternative peers source usage. Why not make an explicit switch?
```suggestion
    boost::optional<PeersList> opt_alternative_peers,
```

Reply:
by the way, we could move this defenition to somewhere at `shared_model/interfaces/...` and reuse it
https://github.com/hyperledger/iroha/blob/6c77597ab55163667931591e774f6a04d91b403b/irohad/ametsuchi/ledger_state.hpp#L15